docs/advanced/components/components.md

```diff
@@ -72,13 +72,16 @@ created () {
 
 ### [Vuex](/stores/vuex/vuex/)
 
-Using Vuex you're implementing the Flux architecture
+Using Vuex you're implementing the Flux architecture.
 
 ### Dependency Injection
 
-Allows you to define a service via the `provide` property, which should be an object or a function that returns an object, and make it available to all of the component's descendants, not juts it's direct children.
+There are two alternative Vue.js handles dependency injection modules:
 
-Please avoid using `this.$root` because can creates too many dependencies.
+- vue-component: is very lightweight DI framework which allows you to register and manually or automatically inject anything anywhere.
+- vue-inject: is a bit more functional, it allows all what vue-component do. But also it does introduce simple service/factory/constant implementation with support of automatic injection (as argumens) of previously registered entities.
```

Comment:
I think we can reword this in a similar fashion:

_But also it does introduce simple service/factory/constant implementation with automatic injection support (as arguments) for previously registered entities._

**arguments** was misspelled.